### DISCUSSION TITLE
chrome warnings on forms and labels

### DISCUSSION THREAD

**Discussion by logan858** [videojs/video.js#8430] [Q&A]
everytime my react app initializes a videojs element i get some chrome warnings around labels:
![image](https://github.com/videojs/video.js/assets/78234668/6bfe9b37-51f2-401c-95d2-30f6f44f1d8f)

the 'violating node' is this: 
![image](https://github.com/videojs/video.js/assets/78234668/9b7acbdd-1126-44e6-b97d-39ed58723bda)
i believe it is because the select elements from videojs have no id/name properties.  

and then the second warning:
![image](https://github.com/videojs/video.js/assets/78234668/1bb234e0-3eca-4291-80d7-f027b859ed38)
 coming from multiple elements, here's one: 
![image](https://github.com/videojs/video.js/assets/78234668/8c6ce0d0-2fc0-465b-96df-a2d861d5ab4a)

im wondering if in react i could fix these label issue by attaching specific id/name attributes or update these label associations?

**Comment by mister-ben** [comment]
I noticed this yesterday too. Guess Chrome has recently introduced this check.

I think Chrome is incorrect here though, according to https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA9 _The value of the aria-labelledby attribute is then a space-separated list of all ids in the order in which the label strings referenced should be read by screen readers._ The `<select>` element is labelled by the two `<label>`s with "Text" and "Color"

**Reply by logan858** [reply]
what are these chrome devs even doing.  that makes sense, thanks for the explanation @mister-ben

**Comment by OwenEdwards** [comment]
Technically, from an HTML perspective, the `<label>` elements aren't labels, they're just `<span>` (or `<div>`?) elements, since they don't have a form field as a child and they don't have the `for` attribute.

We _could_ fix this warning by having each `<label>` associated with the respective `<select>` element with `for` and `id`, and then the aria-labelledby would override that labeling to include the two labels. In theory, this would also give a fallback behavior for browsers that don't support ARIA, but we would want to consider how to fallback for every component that uses ARIA - there's no point in having a fallback for just this component.

Alternatively, we could change these `<label>` elements to `<span>`s, but it would break existing CSS styling for video.js.
